plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -172,10 +172,28 @@ func (r *ReferenceManager) ensureBindingReferences(secretNamespace, secretName s
 		return err
 	}
 
+	var (
+		secretQuotaCount  int
+		projectQuotaCount int
+		quotaScopeSecret  = string(garden.QuotaScopeSecret)
+		quotaScopeProject = string(garden.QuotaScopeProject)
+	)
+
 	for _, quotaRef := range quotaRefs {
-		if _, err := r.quotaLister.Quotas(quotaRef.Namespace).Get(quotaRef.Name); err != nil {
+		quota, err := r.quotaLister.Quotas(quotaRef.Namespace).Get(quotaRef.Name)
+		if err != nil {
 			return err
 		}
+		quotaScope := string(quota.Spec.Scope)
+		if quotaScope == quotaScopeProject {
```

Comment:
Why do  you need to convert this to `string`? `quotaScope` and `garden.QuotaScope[Secret|Project]` should be of the same type?

Reply:
yes it is the same type and conversion is not required. Refactored that.